bvm/ballerina-core/src/main/java/org/ballerinalang/bre/BLangCallableUnitCallback.java

```diff
@@ -52,6 +52,7 @@ public BLangCallableUnitCallback(Context nativeCallCtx, WorkerExecutionContext p
     public void notifySuccess() {
         BLangVMUtils.populateWorkerDataWithValues(this.parentCtx.workerLocal, this.retRegs,
                 this.nativeCallCtx.getReturnValues(), this.retTypes);
+        BLangScheduler.handleInterruptibleAfterCallback(this.parentCtx);
```

Comment:
Shouldn't we handle failure scenario in notifyFailure? (example http timeout etc.)

Reply:
Fixed.